### ISSUE TITLE
Highcharts server stopped working.

### ISSUE THREAD

**Issue by jarrettj** [highcharts/node-export-server#314]
#### Expected behaviour
We have been using this server to generate graphs to png for some time. Lately the generation has stopped working. The command we execute is as follows:
```
highcharts-export-server -infile tmp/test.json -outfile tmp/test.png --scale 2 --allowCodeExecution true
```

#### Actual behaviour
The image generated asks to reinstall module:
![image](https://user-images.githubusercontent.com/777547/142157467-3a1a15b9-26c2-44ef-aa67-59abf85d423a.png)

We manually follow the steps on the server and receive all green and good to go:
```
[root@ip-10-0-14-19 highcharts-export-server]# node build.js 
Including maps support
Including Gantt support
Pulling Highcharts from CDN (latest)..
   https://code.highcharts.com/stock/highstock.js
   https://code.highcharts.com/highcharts-more.js
   https://code.highcharts.com/modules/exporting.js
   https://code.highcharts.com/highcharts-3d.js
   https://code.highcharts.com/modules/data.js
   https://code.highcharts.com/modules/funnel.js
   https://code.highcharts.com/modules/solid-gauge.js
   https://code.highcharts.com/modules/heatmap.js
   https://code.highcharts.com/modules/treemap.js
   https://code.highcharts.com/modules/sunburst.js
   https://code.highcharts.com/modules/xrange.js
   https://code.highcharts.com/modules/streamgraph.js
   https://code.highcharts.com/modules/sankey.js
   https://code.highcharts.com/modules/tilemap.js
   https://code.highcharts.com/modules/histogram-bellcurve.js
   https://code.highcharts.com/modules/bullet.js
   https://code.highcharts.com/modules/organization.js
   https://code.highcharts.com/modules/funnel3d.js
   https://code.highcharts.com/modules/pyramid3d.js
   https://code.highcharts.com/modules/dependency-wheel.js
   https://code.highcharts.com/modules/item-series.js
   https://code.highcharts.com/modules/timeline.js
   https://code.highcharts.com/modules/pareto.js
   https://code.highcharts.com/modules/coloraxis.js
   https://code.highcharts.com/modules/venn.js
   https://code.highcharts.com/modules/dumbbell.js
   https://code.highcharts.com/modules/lollipop.js
   https://code.highcharts.com/modules/wordcloud.js
   https://code.highcharts.com/modules/annotations.js
   https://code.highcharts.com/maps/modules/map.js
   https://code.highcharts.com/modules/gantt.js
Creating export template export..
Including styled mode support
   https://code.highcharts.com/stock/js/highstock.js
   https://code.highcharts.com/js/highcharts-more.js
   https://code.highcharts.com/maps/js/modules/exporting.js
   using cached fetch for https://code.highcharts.com/highcharts-3d.js
   using cached fetch for https://code.highcharts.com/modules/data.js
   using cached fetch for https://code.highcharts.com/modules/funnel.js
   using cached fetch for https://code.highcharts.com/modules/solid-gauge.js
   using cached fetch for https://code.highcharts.com/modules/heatmap.js
   using cached fetch for https://code.highcharts.com/modules/treemap.js
   using cached fetch for https://code.highcharts.com/modules/sunburst.js
   using cached fetch for https://code.highcharts.com/modules/xrange.js
   using cached fetch for https://code.highcharts.com/modules/streamgraph.js
   using cached fetch for https://code.highcharts.com/modules/sankey.js
   using cached fetch for https://code.highcharts.com/modules/tilemap.js
   using cached fetch for https://code.highcharts.com/modules/histogram-bellcurve.js
   using cached fetch for https://code.highcharts.com/modules/bullet.js
   using cached fetch for https://code.highcharts.com/modules/organization.js
   using cached fetch for https://code.highcharts.com/modules/funnel3d.js
   using cached fetch for https://code.highcharts.com/modules/pyramid3d.js
   using cached fetch for https://code.highcharts.com/modules/dependency-wheel.js
   using cached fetch for https://code.highcharts.com/modules/item-series.js
   using cached fetch for https://code.highcharts.com/modules/timeline.js
   using cached fetch for https://code.highcharts.com/modules/pareto.js
   using cached fetch for https://code.highcharts.com/modules/coloraxis.js
   using cached fetch for https://code.highcharts.com/modules/venn.js
   using cached fetch for https://code.highcharts.com/modules/dumbbell.js
   using cached fetch for https://code.highcharts.com/modules/lollipop.js
   using cached fetch for https://code.highcharts.com/modules/wordcloud.js
   using cached fetch for https://code.highcharts.com/modules/annotations.js
   using cached fetch for https://code.highcharts.com/maps/modules/map.js
   using cached fetch for https://code.highcharts.com/modules/gantt.js
Creating export template export_styled..
All done! Happy charting!
For documentation, see https://github.com/highcharts/node-export-server
```

However it still does not generate after executing the build command.

The error we receive from generation is as follows:
```
Wed Nov 17 2021 07:42:02 GMT+0000 (Coordinated Universal Time) [error] phantom worker 1 unexpected data - SyntaxError: Expected token ')'

  phantomjs://code/worker.js:658 in loop
{"filename":"tmp/test.png"} {
  data: {
    allowCodeExecution: 'true',
    width: false,
    callback: false,
    resources: false,
    scale: '2',
    constr: 'Chart',
    chart: '{\n' +
      '    colors: [ "#7cb5ec", "#90ed7d", "#f7a35c", "#8085e9", "#9d3e50", "#e4d354", "#2b908f", "#f45b5b", "#91e8e1", "#434348" ],\n' +
      '    credits: {\n' +
      '        enabled: false\n' +
      '    },\n' +
      '    chart: {\n' +
      '        plotBackgroundColor: null,\n' +
      '        plotShadow: false,\n' +
      "        type: 'pie',\n" +
      "        borderColor: '#808080',\n" +
      '        borderWidth: 1,\n' +
      '    },\n' +
      '    title: {\n' +
      "        text: 'Total Number of Risks by Category'\n" +
      '    },\n' +
      '    accessibility: {\n' +
      '        point: {\n' +
      "            valueSuffix: '%'\n" +
      '        }\n' +
      '    },\n' +
      '    plotOptions: {\n' +
      '        pie: {\n' +
      '            dataLabels: {\n' +
      '                enabled: true,\n' +
      "                format: '<b>{point.name}</b>: {point.y} ({point.percentage:.1f} %)'\n" +
      '            }\n' +
      '        }\n' +
      '    },\n' +
      '    series: [{\n' +
      "        name: 'Category',\n" +
      '        colorByPoint: true,\n' +
      '        data: [{"name":"Litigation","y":2},{"name":"Employment \\/ Gig","y":1},{"name":"M&A \\/ Antitrust","y":1},{"name":"Privacy","y":2},{"name":"Compliance","y":5},{"name":"Regulatory ","y":3},{"name":"GR","y":1}]    }],\n' +
      '    exporting: {\n' +
      '        enabled: false\n' +
      '    }\n' +
      '}',
    svgstr: undefined,
    format: 'png',
    out: 'tmp/test.png',
    styledMode: false,
    asyncRendering: undefined,
    async: true,
    reqID: '',
    globalOptions: false,
    themeOptions: false,
    customCode: false,
    dataOptions: false,
    id: 1
  },
  working: true,
  ready: false,
  alive: true,
  process: ChildProcess {
    _events: [Object: null prototype] {
      uncaughtException: [Function],
      error: [Function],
      close: [Function]
    },
    _eventsCount: 3,
    _maxListeners: undefined,
    _closesNeeded: 3,
    _closesGot: 0,
    connected: false,
    signalCode: null,
    exitCode: null,
    killed: false,
    spawnfile: '/home/node/.npm-global/lib/node_modules/highcharts-export-server/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs',
    _handle: Process {
      onexit: [Function],
      pid: 457,
      [Symbol(owner_symbol)]: [Circular]
    },
    spawnargs: [
      '/home/node/.npm-global/lib/node_modules/highcharts-export-server/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs',
      '/home/node/.npm-global/lib/node_modules/highcharts-export-server/lib/../phantom/worker.js',
      '/home/node/.npm-global/lib/node_modules/highcharts-export-server/lib/../'
    ],
    pid: 457,
    stdin: Socket {
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: null,
      _readableState: [ReadableState],
      readable: false,
      _events: [Object: null prototype],
      _eventsCount: 1,
      _maxListeners: undefined,
      _writableState: [WritableState],
      writable: true,
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: null,
      _server: null,
      [Symbol(asyncId)]: 8,
      [Symbol(kHandle)]: [Pipe],
      [Symbol(kSetNoDelay)]: false,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0
    },
    stdout: Socket {
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: null,
      _readableState: [ReadableState],
      readable: true,
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _writableState: [WritableState],
      writable: false,
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: null,
      _server: null,
      [Symbol(asyncId)]: 9,
      [Symbol(kHandle)]: [Pipe],
      [Symbol(kSetNoDelay)]: false,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0
    },
    stderr: Socket {
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: null,
      _readableState: [ReadableState],
      readable: true,
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _writableState: [WritableState],
      writable: false,
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: null,
      _server: null,
      [Symbol(asyncId)]: 10,
      [Symbol(kHandle)]: [Pipe],
      [Symbol(kSetNoDelay)]: false,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0
    },
    stdio: [ [Socket], [Socket], [Socket] ],
    [Symbol(kCapture)]: false
  },
  workStartTime: 1637134917464,
  ondone: [Function],
  workcount: 25,
  incoming: "SyntaxError: Expected token ')'\n" +
    '\n' +
    '  phantomjs://code/worker.js:658 in loop\n' +
    '{"filename":"tmp/test.png"}',
  id: 1,
  doDone: [Function: doDone],
  restart: [Function],
  work: [Function],
  allowRestart: true
}
```

#### Reproduction steps
 
Using the latest version of node installed using nvm. 
```
[ec2-user@ip-10-0-14-19 ~]$ node -v
v17.1.0
```

Installed highcharts-export-server globally.
```
echo yes | npm install highcharts-export-server -g --unsafe-perm
```

Test data:
```
{
    colors: [ "#7cb5ec", "#90ed7d", "#f7a35c", "#8085e9", "#9d3e50", "#e4d354" ],
    credits: {
        enabled: false
    },
    chart: {
        plotBackgroundColor: null,
        plotShadow: false,
        type: 'pie',
        borderColor: '#808080',
        borderWidth: 1,
    },
    title: {
        text: 'Example graph'
    },
    accessibility: {
        point: {
            valueSuffix: '%'
        }
    },
    plotOptions: {
        pie: {
            dataLabels: {
                enabled: true,
                format: '<b>{point.name}</b>: {point.y} ({point.percentage:.1f} %)'
            }
        }
    },
    series: [{
        name: 'Category',
        colorByPoint: true,
        data: [{"name":"Litigation","y":123},{"name":"Employment \/ Gig","y":321},{"name":"M&A \/ Antitrust","y":111},{"name":"Privacy","y":2}]    }],
    exporting: {
        enabled: false
    }
}

```

We are running on an Amazon Linux 1 server. Not sure if that makes any difference.

Any help would be appreciated. Thanks.

Regards.
Jarrett

**Comment by spiftire** [comment]
I started getting a similar error after highcharts was updated to 9.3.0
Seems that the export server always fetches the latest version of highcharts. I've been experimenting with setting `HIGHCHARTS_VERSION=v9.2.2 ` without any promising results.

**Comment by jarrettj** [comment]
@spiftire 

Thanks for pointing me in the correct direction. I would never have thought of changing the downloaded highcharts version.

Assumed that that was pegged to the node-export-server package version. Which it is clearly not. The version that works for me is 9.2.2 anything after that fails. 

Leaving the ticket open as it might be a bug in the latest version of highcharts.

**Comment by spiftire** [comment]
@jarrettj 
Good that you got it working again. I just gave it another shot myself and got it working with 9.2.2 also. Hopefully the announced new release will fix this.

**Comment by tetzng** [comment]
I was getting the same error with 9.3.3, so I changed to 9.2.2 and that fixed it.

**Comment by tobwen** [comment]
`HIGHCHARTS_VERSION=v9.2.2` also fixed it for me. Thanks!

**Comment by alberchou** [comment]
Thank you all! 

I was having the same problem making experiments with highcharts export server and it was fixed building it with highcharts version 9.2.2 (by default it uses latest one)

**Comment by jakubSzuminski** [comment]
Closing this issue as the problem was related to a certain Highcharts version.

---

I encourage you to try out our latest Puppeteer-based version of the Export Server (here’s a short setup guide: https://github.com/highcharts/node-export-server#puppeteer-version-information).

If you encounter any problems in the newest version of the Export Server, please report it and we will investigate it.
